GitHub.Collectors.Authentication/GitHubAppAuthenticationBase.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Azure.Core;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.CloudMine.GitHub.Collectors.Authentication
+{
+    /// <summary>
+    /// Adding this abstract class to align GitHub collector and CodeAsData authentication workflow.
+    /// CodeAsData will have an implementation with Octokit, while GitHub collector use its own underlying implementation.
+    /// </summary>
+    public abstract class GitHubAppAuthenticationBase
+    {
+        protected string organization;
+        protected readonly int appId;
+        protected readonly string gitHubAppKeyVaultUri;
+
+        public GitHubAppAuthenticationBase(string organization, int appId, string gitHubAppKeyVaultUri)
+        {
+            this.organization = organization;
+            this.appId = appId;
+            this.gitHubAppKeyVaultUri = gitHubAppKeyVaultUri; 
+        }
+
+        /// <summary>
+        /// Main function to get the installation token.
+        /// </summary>
+        protected async Task<string> GetInstallationTokenAsync(string jwt)
+        {
+            if (organization == null)
+            {
+                throw new Exception("Organization must has value to get the installation token.");
+            }
+
+            string installationId = await FindInstallationId(jwt).ConfigureAwait(false);
+            string result = await ObtainPat(jwt, installationId).ConfigureAwait(false);
+            return result;
+        }
+
+        public abstract Task<string> GetAuthorizationHeaderAsync();
```

Comment:
Where is this method used?

Reply:
It was called by HttpClientWrapper.cs in GitHub Collector, and will be called in CodeAsData.

Reply:
The reason for the abstract is to have let CodeAsData and GitHub collector have their own implementation for now. But once CodeAsData migrate to be one of the GitHub collector, it is possible that we abandon the current implementation in CodeAsData and use the implementation in GitHub collector.

Reply:
Moved the uncalled method in this class to an interface: IGithHubAppAuthentication.cs